refactor Jenkins e2e tests

kibbles-n-bytes opened the pull request:
Includes some refactoring and minor changes to the Jenkins tests.

1) API Server uses `Nodeport` service type instead of `LoadBalancer`, which makes resource management simpler on GKE and offers a speed-up.
2) Refactors `contrib/hack/test_walkthrough.sh` to use `contrib/jenkins/install_catalog.sh`, which was already in use by `contrib/jenkins/run_e2e.sh` and had a lot of duplication.
3) Moves `contrib/hack/test_walkthrough.sh` to contrib/jenkins`. No one except for me seemed to use it locally anyway.
4) Little cleanups.

NOTE: The Jenkins run for this will fail. It also requires some Jenkins-side changes, which I can't do until this goes in. DO NOT IMMEDIATELY MERGE, but if it gets 2 LGTMs, message me (@mkibbe on Slack) so that I can merge and do the Jenkins-side changes all at once. I ran the tests on my separate Jenkins testing project, so proof of a valid run can be found here: https://service-catalog-jenkins.appspot.com/job/kibbles-n-bytes-testing/311

MHBauer commented on contrib/jenkins/install_catalog.sh:
```diff
@@ -22,62 +22,56 @@ ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
     helm install "${ROOT}/charts/catalog" \
-    --name "${CATALOG_RELEASE}" \
+    --name "catalog" \
     --namespace "catalog" \
```
No action needed, unless you want to do it.

I think this is the only place we test whether the helm chart works. Can we randomize these to make sure the namespaces haven't been hardcoded or confused? 8char lowercase string should do it. Heck, even one chara to make them collide every once in a while to see what's up.

kibbles-n-bytes replied:
That makes sense, and yep this is the only place the helm chart is actually validated. I can do this, though I'd prefer in a follow-up.

MHBauer replied:
follow up would be great. just wanted to float the idea since I saw that variable and what those flags were set to.

pmorie replied:
I'm cool with @MHBauer's idea, as long as we have a log line that unambiguously identifies what namespace is being used.

kibbles-n-bytes commented:
@MHBauer Sure, I'll make it explicitly set the API Server service type.

Nothing too terribly exciting, mostly lots of little things. I switched from a second kubeconfig for the catalog to a second context, the `setup-sc-context.sh` script was changed to grab the node's IP instead of the apiserver service IP, switched booleans to use `true` for more readability...

pmorie commented on contrib/jenkins/install_catalog.sh:
```diff
@@ -112,17 +108,13 @@ wait_for_expected_output -e 'catalog-catalog-controller-manager' \
+echo 'Waiting for Service Catalog API Server to be up...'
 
-echo 'Connecting to service catalog API Server...'
+${ROOT}/contrib/jenkins/setup-sc-context.sh \
```
I cannot wait until we get support for the aggregator in GKE

kibbles-n-bytes replied:
Same :sweat_smile:

pmorie commented on contrib/jenkins/run_e2e.sh:
```diff
@@ -94,7 +92,9 @@ ${ROOT}/contrib/jenkins/install_catalog.sh ${ARGUMENTS} \
   || error_exit "Error when making e2e test binary."
 
-KUBECONFIG="${KUBECONFIG}" SERVICECATALOGCONFIG="${SC_KUBECONFIG}" ${ROOT}/bin/e2e.test \
+KUBECONFIG="${KUBECONFIG}" ${ROOT}/bin/e2e.test \
```
Hrm, is this using the aggregator now?

kibbles-n-bytes replied:
Nope, it's not, the service catalog context stuff is split up onto the lines below because it got too long.

pmorie commented on contrib/jenkins/install_catalog.sh:
```diff
@@ -22,62 +22,58 @@ ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
     kubectl --namespace kube-system get configmap extension-apiserver-authentication \
   || error_exit 'Timed out waiting for extension-apiserver-authentication configmap to come up.'
 
 # The API server automatically provisions the configmap, but we need it to contain the requestheader CA as well,
```
Nit, and totally fine for a follow-up.  It's not super-clear what 'the configmap' is.

pmorie commented:
LGTM
